src/trio/_highlevel_socket.py

```diff
@@ -412,3 +415,76 @@ async def aclose(self) -> None:
         """Close this listener and its underlying socket."""
         self.socket.close()
         await trio.lowlevel.checkpoint()
+
+
+@final
+class UnixSocketListener(Listener[SocketStream]):
+    """A :class:`~trio.abc.Listener` that uses a listening socket to accept
+    incoming connections as :class:`SocketStream` objects.
```

Comment:
I notice we seem to duplicate the above. Something we could do is avoid exporting these classes and instead require something like

```py
listener = trio.into_listener(socket)
```

This would mean we don't need a `@final` and can potentially provide a smaller API. (I'd like someone with more experience with trio/networking tell me if this is a decent idea or not)

Reply:
Alternatively, we can make `SocketListener` support unix sockets trivially: I assume there's a way to detect this for a socket object. All that changes is:
 - the `socket.type` check
 - `aclose` has an extra `unlink`
 - some docs maybe

Then we don't even need to check for whether unix sockets are supported -- if `SocketListener` is passed one, then obvioiusly they're supported!

Reply:
Added path functionality to `SocketListener` in c922a52